front/lib/resources/labs_transcripts_resource.ts

```diff
@@ -140,6 +140,14 @@ export class LabsTranscriptsConfigurationResource extends BaseResource<LabsTrans
     return this.update({ isActive });
   }
 
+  async setDataSourceId(dataSourceId: number | null) {
```

Comment:
can we do setDataSource(dataSourceId: string | null) :pray: 

We have sIds on data sources now (even if their value is the name for now)

Reply:
I don't see any `sId` column in the `data_sources` table - so when you mean their value is the name, it means that we're actually using `datasource.name` as an sId?

Reply:
Ok so the sId is not materialized yet but it does exist on DataSourceType (and is equal to name)

Let's do that:
- Use a string in the endpoint for dataSourceId (this will be the DataSourceType.sid which is the name for now but will be an sId in the future)
- Add a fetchById on DataSourceResource which simply call fetchByName
- Use a string here and use that new fetchById in the implementation

cc @flvndvd

Reply:
Done